src/components/AuthPanel/AuthPanel.tsx

```diff
@@ -55,48 +78,35 @@ const AuthPanel: FC = () => {
   );
 
   const loginBtn = (
-    <IconButton aria-label="login-icon" onClick={() => handleOpenModal(ModalType.SIGNIN)}>
+    <IconButton aria-label="login-icon" onClick={handleOpenSignInModal}>
       <LoginIcon fontSize="large" />
     </IconButton>
   );
 
-  const handleSignUpFormSumbit = async (data: SignUpSchemaType): Promise<void> => {
-    const {confirmPassword, ...userData} = data;
-    const result: any = await dispatch(registerUser(userData));
-    if (!result.error) {
-      handleCloseModal();
-    }
-    dispatch(getUser());
-  };
-
-  const handleSignInFormSubmit = async (data: SignInSchemaType): Promise<void> => {
-    const result: any = await dispatch(loginUser(data));
-    if (!result.error) {
-      handleCloseModal();
-    }
-    dispatch(getUser());
-  };
+  useEffect(() => {
+    dispatch(resetErrors());
+  }, [dialogContent]);
 
   return (
     <div>
       {user ? userBtn : loginBtn}
       <UserMenu menuItems={menuItems} anchorEl={anchorEl} open={openUserMenu} handleClose={handleCloseUserMenu} />
-      {dialogType !== null && (
-        <BasicDialog
-          title={dialogType === ModalType.SIGNIN ? 'Sign in' : 'Registration'}
-          open={dialogType !== null}
-          onClose={handleCloseModal}
-        >
-          {dialogType === ModalType.SIGNIN ? (
-            <SignInForm
-              onSignUpClick={() => handleOpenModal(ModalType.SIGNUP)}
-              onFormSubmit={handleSignInFormSubmit}
-              error={userError}
-            />
-          ) : (
-            <SignUpForm onFormSubmit={handleSignUpFormSumbit} error={userError} />
-          )}
-        </BasicDialog>
+      {dialogContent !== null && (
+        <>
+          <BasicDialog open={dialogContent !== null} onClose={handleCloseModal}>
+            {dialogContent}
+            {signUpError && (
+              <Box width={300}>
+                <BasicError text={signUpError} />
+              </Box>
+            )}
+            {signInError && (
+              <Box width={300}>
+                <BasicError text={signInError} />
+              </Box>
+            )}
+          </BasicDialog>
```

Comment:
It isn't clear logic for now. Because you should keep your dialog on a root level. It doesn't make sense to save the content in redux and the keep the dialog in each component you need. Let`s discuss it tomorrow.

Reply:
Ok, now I did it another way. No props from parent to Dialog. All actions about Dialog in slice and rendering always dialogContent.